pkg/pool-weighted/test/foundry/FungibilityWeighted.t.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import { ArrayHelpers } from "@balancer-labs/v3-solidity-utils/contracts/test/ArrayHelpers.sol";
+import { CastingHelpers } from "@balancer-labs/v3-solidity-utils/contracts/helpers/CastingHelpers.sol";
+
+import { PoolHooksMock } from "@balancer-labs/v3-vault/contracts/test/PoolHooksMock.sol";
+import { ProtocolFeeControllerMock } from "@balancer-labs/v3-vault/contracts/test/ProtocolFeeControllerMock.sol";
+import { FungibilityTest } from "@balancer-labs/v3-vault/test/foundry/Fungibility.t.sol";
+
+import { WeightedPoolFactory } from "../../contracts/WeightedPoolFactory.sol";
+import { WeightedPool } from "../../contracts/WeightedPool.sol";
+import { WeightedPoolContractsDeployer } from "./utils/WeightedPoolContractsDeployer.sol";
+
+contract FungibilityWeightedTest is WeightedPoolContractsDeployer, FungibilityTest {
+    using ArrayHelpers for *;
+    using CastingHelpers for address[];
+
+    uint256 internal constant DEFAULT_SWAP_FEE = 1e16; // 1%
```

Comment:
Could move DEFAULT_SWAP_FEE up to the parent Test

Reply:
Good point, I'll do it